eng/testing/mobile/WasmRunnerTemplate.sh

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env bash
```

Comment:
It is confusing to have wasm under mobile. wasm != mobile. I would call the directory `custom` or something like that.

Reply:
Or `lightweight`

Reply:
I wondered about that as well and @akoeplinger told me that wasm is declared as mobile: https://github.com/dotnet/runtime/blob/a109254d8afbb0c781fcefea0ef41e22bf42c5e2/eng/Configurations.props#L68.

Reply:
What is the definition of "mobile" in plain English that we are using in this context?

I see `TargetsMobile` used for multiple different meanings that happen to work today, but that are not robust against changes in the platforms matrix. Unless there is non-ambiguous definition of what this means, we should not be promoting use of this name throughout the tree.

Reply:
Maybe we can open an issue for this and bring that discussion there? Changing this on the current PR would make the PR diff way bigger than it is as we already use the `TargetsMobile` property in MSBuild across multiple places in the repo.

Reply:
The usage of the term "mobile" traces back a few years to mono/mono where we essentially had "desktop" and "mobile" for turning on/off certain runtime features and when wasm was brought up it was closer to the existing "mobile" config :)

Reply:
> Maybe we can open an issue for this and bring that discussion there?

Fine with me. This PR is renaming a number of files to mobile. Should that rename be undone until we agree on the right naming? (Undoing the rename should make the PR smaller.)